roles/dnsmasq/defaults/main.yml

```diff
@@ -33,3 +30,46 @@ dnsmasq_min_replicas: 1
 
 # Custom name servers
 dnsmasq_upstream_dns_servers: []
+
+# Try each query with each server strictly in the order
+dnsmasq_enable_strict_order: false
```

Comment:
`dnsmasq_enable_strict_order` should be set to true by default in order to fit the current behaviour